### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by Konstantin35** [etclabscore/go-etchash#4]
When using 
`var hasher = etchash.New(nil, nil)`
after sending the first solution, this error occurs and the pool is disconnected

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x750440]
goroutine 8070 [running]:
github.com/etclabscore/go-etchash.(*Light).getCache(0xc00018ee80, 0x731031, 0x1)
        /pool/build/_workspace/src/github.com/etclabscore/go-etchash/etchash.go:513 +0x480
github.com/etclabscore/go-etchash.(*Light).Verify(0xc00018ee80, 0x898300, 0xc0000ffb60, 0xc0000ffb60)
        /pool/build/_workspace/src/github.com/etclabscore/go-etchash/etchash.go:446 +0x10a
github.com/sammy007/open-ethereum-pool/proxy.(*ProxyServer).processShare(0xc00010c280, 0xc000bdcdb0, 0x2a, 0xc000bdcdab, 0x5, 0xc001740ea0, 0xd, 0xc000168000, 0xc00018ee40, 0x3, ...)
        /pool/build/_workspace/src/github.com/sammy007/open-ethereum-pool/proxy/miner.go:53 +0x4aa
github.com/sammy007/open-ethereum-pool/proxy.(*ProxyServer).handleSubmitRPC(0xc00010c280, 0xc0017a6000, 0xc000bdcdb0, 0x2a, 0xc000bdcdab, 0x5, 0xc00018ee40, 0x3, 0x4, 0x79d7a0, ...)
        /pool/build/_workspace/src/github.com/sammy007/open-ethereum-pool/proxy/handlers.go:88 +0x1ab
github.com/sammy007/open-ethereum-pool/proxy.(*ProxyServer).handleTCPSubmitRPC(0xc00010c280, 0xc0017a6000, 0xc000015410, 0x5, 0xc00018ee40, 0x3, 0x4, 0xe9, 0x0)
        /pool/build/_workspace/src/github.com/sammy007/open-ethereum-pool/proxy/handlers.go:69 +0xee
github.com/sammy007/open-ethereum-pool/proxy.(*Session).handleTCPMessage(0xc0017a6000, 0xc00010c280, 0xc00040a6e0, 0x7928c0, 0xc00040a6e0)
        /pool/build/_workspace/src/github.com/sammy007/open-ethereum-pool/proxy/stratum.go:700 +0x5fc
github.com/sammy007/open-ethereum-pool/proxy.(*ProxyServer).handleTCPClient(0xc00010c280, 0xc0017a6000, 0xc0016eb7b8, 0x75a21f)
        /pool/build/_workspace/src/github.com/sammy007/open-ethereum-pool/proxy/stratum.go:663 +0x27f
github.com/sammy007/open-ethereum-pool/proxy.(*ProxyServer).ListenTCP.func3(0xc00010c280, 0xc0016f6030, 0x899e40, 0xc0004020e0, 0xc000626000, 0xc0017a6000)
        /pool/build/_workspace/src/github.com/sammy007/open-ethereum-pool/proxy/stratum.go:625 +0x39
created by github.com/sammy007/open-ethereum-pool/proxy.(*ProxyServer).ListenTCP
        /pool/build/_workspace/src/github.com/sammy007/open-ethereum-pool/proxy/stratum.go:624 +0x670
```

**Comment by masnail** [comment]
use nullptr point,    in etchash.go:513
fix it:
https://github.com/masnail/go-etchash/commit/a3c0dafc38bfea02cf789c26a8f5ce16bdf3c1f8

**Comment by iquidus** [comment]
possibly fixed by #6

**Comment by nottug** [comment]
Didn't see this but yeah its identical, should be fixed
